waiter/test/waiter/auth/saml_test.clj

```diff
@@ -0,0 +1,190 @@
+                                 (merge {:request-method :post
+                                         :body (StringBufferInputStream. "saml-auth-data=my-saml-auth-data")}))]
+          (is (= {:authorization/principal "my-user@domain"
+                  :authorization/user "my-user"
+                  :body ""
+                  :headers {"location" "redirect-url"
+                            "set-cookie" "x-waiter-auth=%5B%22my%2Duser%40domain%22+1557792000000%5D;Max-Age=86400;Path=/;HttpOnly=true"}
+                  :status 303}
+                 (saml-auth-redirect-handler dummy-request' saml-authenticator))))))
+    (testing "has saml-auth-data short expiry"
+      (with-redefs [b64/decode identity
+                    b64/encode identity
+                    nippy/freeze (fn [data _] (.getBytes (str data)))
+                    nippy/thaw (fn [data _]
+                                 (if (= "my-saml-auth-data" (String. data))
+                                   {:not-on-or-after (t/plus test-time (t/hours 1)) :saml-principal "my-user@domain" :redirect-url "redirect-url"}
+                                   nil))
+                    t/now (fn [] test-time)]
+        (let [dummy-request' (-> (merge-with merge dummy-request {:headers {"content-type" "application/x-www-form-urlencoded"}})
+                                 (merge {:request-method :post
+                                         :body (StringBufferInputStream. "saml-auth-data=my-saml-auth-data")}))]
+          (is (= {:authorization/principal "my-user@domain"
+                  :authorization/user "my-user"
+                  :body ""
+                  :headers {"location" "redirect-url"
+                            "set-cookie" "x-waiter-auth=%5B%22my%2Duser%40domain%22+1557792000000%5D;Max-Age=3600;Path=/;HttpOnly=true"}
+                  :status 303}
+                 (saml-auth-redirect-handler dummy-request' saml-authenticator))))))))
+
+(defn- normalize-x509-certificate-string
+  [cert-str]
+  (-> cert-str
+      (string/replace #"From: [^,]+" "")
+      (string/replace #"To: [^\]]+" "")))
+
+(deftest test-certificate-x509
+  (is (= (normalize-x509-certificate-string (slurp "test-files/saml/x509-certificate-to-string.txt"))
+         (normalize-x509-certificate-string (str (certificate-x509 (slurp (:idp-cert-uri valid-config))))))))
+
+(defn- saml-response-from-xml
+  [change-user?]
+  (str->deflate->base64
+    (string/replace (slurp "test-files/saml/saml-response.xml")
+                    "user1@example.com"
+                    (str (if change-user? "root" "user1") "@example.com"))))
+(def relay-state-string
+  (utils/map->base-64-string {:host "host" :request-url "request-url" :scheme "scheme"} [:salted "password"]))
+
+(deftest test-saml-acs-handler
+  (let [saml-authenticator (dummy-saml-authenticator)
+        test-time (clj-time.format/parse "2019-05-14")
+        processed-saml-response {:body "<!doctype html>
```

Comment:
I would mock the response generator and validate the return response as a map instead of checking the raw string.

Reply:
done